src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskExceptionHolder.cs

```diff
@@ -277,17 +277,16 @@ internal AggregateException CreateExceptionObject(bool calledFromFinalizer, Exce
         }
 
         /// <summary>
-        /// Wraps the exception dispatch infos into a new read-only collection. By calling this method,
-        /// the holder assumes exceptions to have been "observed", such that the finalization
+        /// The holder assumes exceptions to have been "observed", such that the finalization
         /// check will be subsequently skipped.
         /// </summary>
-        internal ReadOnlyCollection<ExceptionDispatchInfo> GetExceptionDispatchInfos()
+        internal List<ExceptionDispatchInfo> GetExceptionDispatchInfos()
         {
             List<ExceptionDispatchInfo>? exceptions = m_faultExceptions;
             Debug.Assert(exceptions != null, "Expected an initialized list.");
             Debug.Assert(exceptions.Count > 0, "Expected at least one exception.");
             MarkAsHandled(false);
-            return new ReadOnlyCollection<ExceptionDispatchInfo>(exceptions);
+            return exceptions!;
```

Comment:
Why is the `!` needed here?

Reply:
The null check is done only inside Assert 3 lines above if that's what you mean

Reply:
Yes.  The assert should mean that the `!` isn't needed.  Are you getting a compiler warning/error without the `!`?

Reply:
Hmm, I think I was warning from it but cannot repro it again. Removing it